### ISSUE TITLE
Adaptive Card Toggle.Visibility action does not work

### ISSUE THREAD

**Issue by michaelslz23** [MicrosoftDocs/msteams-docs#8157]
It just happen in recent days that Adaptive Card Toggle.Visibility action does not work anymore.

**Comment by Prasad-MSFT** [comment]
@michaelslz23 - Could you please share your card JSON which you are using, so that we can test it once from our end?

**Comment by michaelslz23** [comment]
Here's one of the .json we are using, thanks.  

`{
    "type": "AdaptiveCard",
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.4",
    "msTeams": {
        "width": "full"
    },
    "body": [
        {
            "type": "ColumnSet",
            "columns": [
                {
                    "type": "Column",
                    "width": "auto",
                    "items": [
                        {
                            "type": "TextBlock",
                            "wrap": true,
                            "text": "Astro Alert -",
                            "size": "Large",
                            "weight": "Bolder",
                            "color": "Accent"
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "Dec 2022",
                            "wrap": true,
                            "size": "Large",
                            "weight": "Bolder",
                            "color": "Accent"
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "auto",
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "2023-01-03",
                            "wrap": true,
                            "size": "Medium"
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "style": "emphasis",
            "items": [
                {
                    "type": "ColumnSet",
                    "columns": [
                        {
                            "type": "Column",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "wrap": true,
                                    "weight": "Bolder",
                                    "text": "LOB"
                                }
                            ],
                            "width": 11
                        },
                        {
                            "type": "Column",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "wrap": true,
                                    "weight": "Bolder",
                                    "text": "DP"
                                }
                            ],
                            "width": 13
                        },
                        {
                            "type": "Column",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "METRIC",
                                    "wrap": true,
                                    "weight": "Bolder"
                                }
                            ],
                            "width": 32
                        },
                        {
                            "type": "Column",
                            "width": 15,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "wrap": true,
                                    "text": "VALUE",
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 18,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "NEXT MTH",
                                    "wrap": true,
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "weight": "Bolder",
                                    "text": "TARGET"
                                }
                            ]
                        }
                    ]
                }
            ],
            "bleed": true
        },
        {
            "type": "Container",
            "items": [
                {
                    "type": "ColumnSet",
                    "columns": [
                        {
                            "type": "Column",
                            "width": 12,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Partner",
                                    "weight": "Bolder",
                                    "wrap": true
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 14,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Concentrix",
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 36,
                            "style": "default",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "FrontLine - All Up CSAT",
                                    "wrap": true
                                },
                                {
                                    "type": "TextBlock",
                                    "text": " ",
                                    "wrap": true,
                                    "fontType": "Monospace",
                                    "color": "Warning"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 15,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.59",
                                    "wrap": true,
                                    "weight": "Bolder",
                                    "color": "Attention",
                                    "size": "Large"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 16,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.63 (4.47-4.79)",
                                    "wrap": true,
                                    "color": "Attention"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.65",
                                    "wrap": true,
                                    "horizontalAlignment": "Right"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "auto",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://adaptivecards.io/content/down.png",
                                    "width": "20px",
                                    "id": "Down0"
                                },
                                {
                                    "type": "Image",
                                    "url": "https://adaptivecards.io/content/up.png",
                                    "width": "20px",
                                    "isVisible": false,
                                    "id": "Up0"
                                }
                            ],
                            "selectAction": {
                                "type": "Action.ToggleVisibility",
                                "title": "expand",
                                "targetElements": [
                                    "Content0",
                                    "Down0",
                                    "Up0"
                                ]
                            }
                        }
                    ]
                },
                {
                    "type": "Container",
                    "isVisible": false,
                    "id": "Content0",
                    "items": [
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": 9
                                },
                                {
                                    "type": "Column",
                                    "width": 78,
                                    "items": [
                                        {
                                            "type": "TextBlock",
                                            "text": "**6** time(s) missed target in 12 months.",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "DP/Site:\r - **Teleperformance Lisbon (PT) ~ 4.4** (**6 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Issue Code:\r - **Billing ~ 4.15** (**5 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "PCY:\r - **CSP Delivery ~ 4.4** (**6 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "items": [
                {
                    "type": "ColumnSet",
                    "columns": [
                        {
                            "type": "Column",
                            "width": 12,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Partner",
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 14,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Teleperformance",
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 36,
                            "style": "default",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "FrontLine - All Up CSAT",
                                    "wrap": true
                                },
                                {
                                    "type": "TextBlock",
                                    "text": " ",
                                    "wrap": true,
                                    "fontType": "Monospace",
                                    "color": "Warning"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 15,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.44",
                                    "wrap": true,
                                    "weight": "Bolder",
                                    "color": "Attention",
                                    "size": "Large"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 16,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.47 (4.21-4.72)",
                                    "wrap": true,
                                    "color": "Attention"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.65",
                                    "wrap": true,
                                    "horizontalAlignment": "Right"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "auto",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://adaptivecards.io/content/down.png",
                                    "width": "20px",
                                    "id": "Down1"
                                },
                                {
                                    "type": "Image",
                                    "url": "https://adaptivecards.io/content/up.png",
                                    "width": "20px",
                                    "isVisible": false,
                                    "id": "Up1"
                                }
                            ],
                            "selectAction": {
                                "type": "Action.ToggleVisibility",
                                "title": "expand",
                                "targetElements": [
                                    "Content1",
                                    "Down1",
                                    "Up1"
                                ]
                            }
                        }
                    ]
                },
                {
                    "type": "Container",
                    "isVisible": false,
                    "id": "Content1",
                    "items": [
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": 9
                                },
                                {
                                    "type": "Column",
                                    "width": 78,
                                    "items": [
                                        {
                                            "type": "TextBlock",
                                            "text": "**11** time(s) missed target in 12 months.",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "PCY:\r - **MPN Delivery ~ 4.27** (**3 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "DP/Site:\r - **Teleperformance Lisbon (PT) ~ 4.4** (**6 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Area & Region:\r - **France ~ 4.24** (**6 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Issue Code:\r - **Membership ~ 3.92** (**1 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "\r - **Account Settings ~ 4.17** (**2 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "items": [
                {
                    "type": "ColumnSet",
                    "columns": [
                        {
                            "type": "Column",
                            "width": 12,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Partner",
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 14,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Concentrix",
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 35,
                            "style": "default",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "FrontLine - MPN CSAT",
                                    "wrap": true
                                },
                                {
                                    "type": "TextBlock",
                                    "text": " ",
                                    "wrap": true,
                                    "fontType": "Monospace",
                                    "color": "Warning"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 15,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.64",
                                    "wrap": true,
                                    "weight": "Bolder",
                                    "color": "Attention",
                                    "size": "Large"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 16,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.64 (4.51-4.78)",
                                    "wrap": true,
                                    "color": "Attention"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.7",
                                    "wrap": true,
                                    "horizontalAlignment": "Right"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "auto",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://adaptivecards.io/content/down.png",
                                    "width": "20px",
                                    "id": "Down2"
                                },
                                {
                                    "type": "Image",
                                    "url": "https://adaptivecards.io/content/up.png",
                                    "width": "20px",
                                    "isVisible": false,
                                    "id": "Up2"
                                }
                            ],
                            "selectAction": {
                                "type": "Action.ToggleVisibility",
                                "title": "expand",
                                "targetElements": [
                                    "Content2",
                                    "Down2",
                                    "Up2"
                                ]
                            }
                        }
                    ]
                },
                {
                    "type": "Container",
                    "isVisible": false,
                    "id": "Content2",
                    "items": [
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": 9
                                },
                                {
                                    "type": "Column",
                                    "width": 78,
                                    "items": [
                                        {
                                            "type": "TextBlock",
                                            "text": "**5** time(s) missed target in 12 months.",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Issue Code:\r - **Account Settings ~ 4.52** (**2 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "DP/Site:\r - **Concentrix Manila (PH) ~ 4.32** (**1 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Area & Region:\r - **India ~ 4.33** (**1 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "items": [
                {
                    "type": "ColumnSet",
                    "columns": [
                        {
                            "type": "Column",
                            "width": 12,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Partner",
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 14,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "Teleperformance",
                                    "weight": "Bolder"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 35,
                            "style": "default",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "FrontLine - MPN CSAT",
                                    "wrap": true
                                },
                                {
                                    "type": "TextBlock",
                                    "text": " ",
                                    "wrap": true,
                                    "fontType": "Monospace",
                                    "color": "Warning"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 15,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.27",
                                    "wrap": true,
                                    "weight": "Bolder",
                                    "color": "Attention",
                                    "size": "Large"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": 16,
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.55 (4.23-4.87)",
                                    "wrap": true,
                                    "color": "Attention"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "stretch",
                            "items": [
                                {
                                    "type": "TextBlock",
                                    "text": "4.7",
                                    "wrap": true,
                                    "horizontalAlignment": "Right"
                                }
                            ]
                        },
                        {
                            "type": "Column",
                            "width": "auto",
                            "items": [
                                {
                                    "type": "Image",
                                    "url": "https://adaptivecards.io/content/down.png",
                                    "width": "20px",
                                    "id": "Down3"
                                },
                                {
                                    "type": "Image",
                                    "url": "https://adaptivecards.io/content/up.png",
                                    "width": "20px",
                                    "isVisible": false,
                                    "id": "Up3"
                                }
                            ],
                            "selectAction": {
                                "type": "Action.ToggleVisibility",
                                "title": "expand",
                                "targetElements": [
                                    "Content3",
                                    "Down3",
                                    "Up3"
                                ]
                            }
                        }
                    ]
                },
                {
                    "type": "Container",
                    "isVisible": false,
                    "id": "Content3",
                    "items": [
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": 9
                                },
                                {
                                    "type": "Column",
                                    "width": 78,
                                    "items": [
                                        {
                                            "type": "TextBlock",
                                            "text": "**5** time(s) missed target in 12 months.",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Issue Code:\r - **Membership ~ 3.92** (**4 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "\r - **MPN Benefits and Competencies ~ 4.2** (**2 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        },
                                        {
                                            "type": "TextBlock",
                                            "text": "Area & Region:\r - **France ~ 3.95** (**3 time(s)** was main driver when missing target)",
                                            "wrap": true
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "Container",
            "items": [
                {
                    "type": "Container",
                    "style": "emphasis",
                    "bleed": true,
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": "*Other metrics in your monitoring list do not meet your configured alert criteria.",
                            "wrap": true,
                            "size": "Small"
                        }
                    ]
                },
                {
                    "type": "ActionSet",
                    "actions": [
                        {
                            "type": "Action.OpenUrl",
                            "title": "Open Astro",
                            "url": "https://aka.ms/astro"
                        },
                        {
                            "type": "Action.OpenUrl",
                            "title": "Manage your Alerts",
                            "url": "https://aka.ms/astroalert"
                        }
                    ]
                }
            ]
        }
    ]
}`

**Comment by jonbagg** [comment]
Pilling on: All of our toggle.Visibility cards have stopped working in the Teams Desktop App.  This feature is still working for us in the Teams web app.

A recent issue (#7626 @Meghana-MSFT ) with the same feature also appeared in the desktop app before being seen in the Team web app.

We have impacted production applications.  Please advise

**Comment by trevhg7** [comment]
@michaelslz23, thank you for quickly flagging. We are having the same issue, and it is affecting our customers' ability to use our products via the Teams platform.

Microsoft Support team members, thank you in advance for your swift attention to this issue.

**Comment by jonbagg** [comment]
FYI @joshatrezolveai, @AlexanderPBrown

**Comment by joshatrezolveai** [comment]
@jonbagg - thanks for the tag! The cofounder of my company reported this as an issue to me about three days ago as well. I checked, and they were working fine for me, in the desktop app and web app - but with his reporting of this issue as well as this thread, I can only imagine it will be wide spread soon enough. 

@Prasad-MSFT, please escelate this to the highest priority. We have well over 50 demo's scheduled this month and this feature is HEAVILY utilized in our product. Additionally, we have hundreds of thousands end users that use this product and in turn this feature. I can not wait another 30-45 days to see this issue resolved. Who is doing the QA on these changes? How does this keep happening? Reference issue #7626.

**Comment by Prasad-MSFT** [comment]
Hello @michaelslz23 , @jonbagg , @trevhg7 , @joshatrezolveai  - We tried to repro this issue at our end using both Teams desktop (Microsoft Teams Version 1.6.00.5081) and web client, however it works perfectly fine for us. Attaching screen recording for reference:

https://user-images.githubusercontent.com/62237649/221565009-7d65a67a-8ed3-40f7-9613-17da5946c3ad.mp4

 However, as this issue is impacting a lot, we have brought this to notice of engineering team notifying them about the severity.

Additionally requesting you to provide MS Teams client version details where you are facing this issue along with tenant details its impacting, so that we can pass this info to engineering team to investigate further.

**Comment by jonbagg** [comment]
@Prasad-MSFT, I was on version 1.6.0.0.1381 of teams desktop.  I just upgraded Teams, though the version # hasn't changed yet, and the toggles are now working for my team and me again.  We'll continue to monitor today to see if the problem appears again and will update.

**Comment by michaelslz23** [comment]
My version is 1.6.00.4472 (64-bit).  Let me try to upgrade the version.

**Comment by michaelslz23** [comment]
My version is 1.6.00.4472 (64-bit), it failed to work for Toggle.Visibility in my Adaptive Card messages.   Tried to upgrade, but Teams just shown my version is already latest. :(

**Comment by adityarana420** [comment]
Even I am facing the same issue. Issue is happening when using "selectAction -> ToggleVisibility".

Here is a very simple json code, here

```

  {
    "type": "AdaptiveCard",
    "body": [
        {
            "type": "Container",
            "id": "toggle_me",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Toggling this text..."
                }
            ]
        },
        {
            "type": "Container",
            "items": [
                {
                    "type": "TextBlock",
                    "text": "Click to toggle using 'selectAction'"
                }
            ],
            "selectAction": {
                "type": "Action.ToggleVisibility",
                "targetElements": [ "toggle_me" ]
            }
        },
        {
            "type": "Container",
            "items": [
                {
                    "type": "ActionSet",
                    "actions": [{
                        "type": "Action.ToggleVisibility",
                        "title": "Click to toggle using 'button'",
                        "targetElements": [ "toggle_me" ]
                    }]
                }
            ]
        }    
    ],
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.4"
}
```

Here toggling using the button is working,  but toggling using "selectAction" is not working. This is a fairly new issue.

**Comment by Prasad-MSFT** [comment]
@michaelslz23 , @adityarana420 - Thanks for sharing version details. We will pass this to engineering team and will update you if we hear anything from them.
Thanks again!

**Comment by KaratLee** [comment]
I have encountered Toggle.Visibility not working on both Chrome and Team app. Issue + 1.

Chrome version: 110.0.5481.177 
Teams App version: 1.6.00.1159
AdaptiveCard version: 1.3

**Comment by jonbagg** [comment]
@Prasad-MSFT,  unfortunately, the issue has returned intermittently for us.  We currently can't use the feature on desktop (windows), but can on desktop (mac) and in the web app.  Can you confirm you've raised this issue with Teams engineering and let us know if they can reproduce this issue?

**Comment by andrekiba** [comment]
@Prasad-MSFT could you please provide an update?
This issue has a high impact for some of our customers.
Is there any way to fix the problem in the meantime?

**Comment by ChetanSharma-msft** [comment]
Hello Everyone- we will check the bug status and will let you know the updates.

**Comment by joseglauber** [comment]
I'm having the same problem and I need an urgent fix.
because it's impacting the operation of 5 thousand users of my application.

Teams App Version 1.6.00.1381 (64 bits). Last Update 2023-02-07.

Adaptive Card:

`{
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "type": "AdaptiveCard",
    "version": "1.5",
    "body": [
        {
            "type": "TextBlock",
            "text": "Pesquisa de Satisfação",
            "size": "Large",
            "color": "Light",
            "weight": "Bolder",
            "horizontalAlignment": "Center"
        },
        {
            "type": "TextBlock",
            "text": "Qual nota você atribui ao autoatendimento?",
            "wrap": true,
            "horizontalAlignment": "Center",
            "size": "Medium",
            "color": "Light"
        },
        {
            "type": "ColumnSet",
            "columns": [
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "Image",
                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-otimo-clr.png",
                            "width": "55px",
                            "height": "49px",
                            "horizontalAlignment": "Center",
                            "id": "img_otimo"
                        },
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": "stretch",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-otimo-pb.png",
                                            "width": "55px",
                                            "height": "49px",
                                            "horizontalAlignment": "Center",
                                            "id": "img_otimo_cinza",
                                            "isVisible": false
                                        }
                                    ]
                                }
                            ],
                            "selectAction": {
                                "type": "Action.ToggleVisibility",
                                "title": "OtimoPB",
                                "targetElements": [
                                    {
                                        "elementId": "txt_motivo",
                                        "isVisible": true
                                    },
                                    {
                                        "elementId": "motivo",
                                        "isVisible": true
                                    },
                                    {
                                        "elementId": "act_btn_otimo",
                                        "isVisible": true
                                    },
                                    {
                                        "elementId": "act_btn_bom",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "act_btn_regular",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "act_btn_ruim",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "act_btn_pessimo",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "img_otimo",
                                        "isVisible": true
                                    },
                                    {
                                        "elementId": "img_bom",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "img_regular",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "img_ruim",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "img_pessimo",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "img_otimo_cinza",
                                        "isVisible": false
                                    },
                                    {
                                        "elementId": "img_bom_cinza",
                                        "isVisible": true
                                    },
                                    {
                                        "elementId": "img_regular_cinza",
                                        "isVisible": true
                                    },
                                    {
                                        "elementId": "img_ruim_cinza",
                                        "isVisible": true
                                    },
                                    {
                                        "elementId": "img_pessimo_cinza",
                                        "isVisible": true
                                    }
                                ]
                            },
                            "spacing": "None"
                        },
                        {
                            "type": "TextBlock",
                            "text": "Ótimo",
                            "wrap": true,
                            "horizontalAlignment": "Center",
                            "color": "Light",
                            "weight": "Bolder"
                        }
                    ],
                    "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Otimo",
                        "targetElements": [
                            {
                                "elementId": "txt_motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_otimo",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_bom",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_regular",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_ruim",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_pessimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_otimo",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_bom",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_regular",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_ruim",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_pessimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_otimo_cinza",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_bom_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_regular_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_ruim_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_pessimo_cinza",
                                "isVisible": true
                            }
                        ]
                    }
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "Image",
                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-bom-clr.png",
                            "width": "55px",
                            "height": "49px",
                            "horizontalAlignment": "Center",
                            "id": "img_bom"
                        },
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": "stretch",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-bom-pb.png",
                                            "width": "55px",
                                            "height": "49px",
                                            "id": "img_bom_cinza",
                                            "isVisible": false,
                                            "horizontalAlignment": "Center"
                                        }
                                    ],
                                    "selectAction": {
                                        "type": "Action.ToggleVisibility",
                                        "targetElements": [
                                            {
                                                "elementId": "txt_motivo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "motivo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "act_btn_otimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_bom",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "act_btn_regular",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_ruim",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_pessimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_otimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_bom",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_regular",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_ruim",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_pessimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_otimo_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_bom_cinza",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_regular_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_ruim_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_pessimo_cinza",
                                                "isVisible": true
                                            }
                                        ]
                                    }
                                }
                            ],
                            "spacing": "None"
                        },
                        {
                            "type": "TextBlock",
                            "text": "Bom",
                            "wrap": true,
                            "horizontalAlignment": "Center",
                            "color": "Light",
                            "weight": "Bolder"
                        }
                    ],
                    "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "targetElements": [
                            {
                                "elementId": "txt_motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_otimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_bom",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_regular",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_ruim",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_pessimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_otimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_bom",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_regular",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_ruim",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_pessimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_otimo_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_bom_cinza",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_regular_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_ruim_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_pessimo_cinza",
                                "isVisible": true
                            }
                        ]
                    }
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "Image",
                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-regular-clr.png",
                            "width": "55px",
                            "height": "49px",
                            "horizontalAlignment": "Center",
                            "id": "img_regular"
                        },
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": "stretch",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "id": "img_regular_cinza",
                                            "horizontalAlignment": "Center",
                                            "width": "55px",
                                            "height": "49px",
                                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-regular-pb.png",
                                            "isVisible": false
                                        }
                                    ],
                                    "selectAction": {
                                        "type": "Action.ToggleVisibility",
                                        "targetElements": [
                                            {
                                                "elementId": "txt_motivo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "motivo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "act_btn_otimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_bom",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_regular",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "act_btn_ruim",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_pessimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_otimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_bom",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_regular",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_ruim",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_pessimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_otimo_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_bom_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_regular_cinza",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_ruim_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_pessimo_cinza",
                                                "isVisible": true
                                            }
                                        ]
                                    }
                                }
                            ],
                            "spacing": "None"
                        },
                        {
                            "type": "TextBlock",
                            "text": "Regular",
                            "wrap": true,
                            "horizontalAlignment": "Center",
                            "color": "Light",
                            "weight": "Bolder"
                        }
                    ],
                    "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "targetElements": [
                            {
                                "elementId": "txt_motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_otimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_bom",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_regular",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_ruim",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_pessimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_otimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_bom",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_regular",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_ruim",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_pessimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_otimo_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_bom_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_regular_cinza",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_ruim_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_pessimo_cinza",
                                "isVisible": true
                            }
                        ]
                    }
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "Image",
                            "width": "55px",
                            "height": "49px",
                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-ruim-clr.png",
                            "horizontalAlignment": "Center",
                            "id": "img_ruim"
                        },
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": "stretch",
                                    "spacing": "None",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "id": "img_ruim_cinza",
                                            "width": "55px",
                                            "height": "49px",
                                            "horizontalAlignment": "Center",
                                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-ruim-pb.png",
                                            "isVisible": false
                                        }
                                    ],
                                    "selectAction": {
                                        "type": "Action.ToggleVisibility",
                                        "targetElements": [
                                            {
                                                "elementId": "txt_motivo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "motivo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "act_btn_otimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_bom",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_regular",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_ruim",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "act_btn_pessimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_otimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_bom",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_regular",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_ruim",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_pessimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_otimo_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_bom_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_regular_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_ruim_cinza",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_pessimo_cinza",
                                                "isVisible": true
                                            }
                                        ]
                                    }
                                }
                            ],
                            "spacing": "None"
                        },
                        {
                            "type": "TextBlock",
                            "text": "Ruim",
                            "wrap": true,
                            "horizontalAlignment": "Center",
                            "color": "Light",
                            "weight": "Bolder"
                        }
                    ],
                    "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "targetElements": [
                            {
                                "elementId": "txt_motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_otimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_bom",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_regular",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_ruim",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_pessimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_otimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_bom",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_regular",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_ruim",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_pessimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_otimo_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_bom_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_regular_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_ruim_cinza",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_pessimo_cinza",
                                "isVisible": true
                            }
                        ]
                    }
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "Image",
                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-pessimo-clr.png",
                            "width": "55px",
                            "height": "49px",
                            "horizontalAlignment": "Center",
                            "id": "img_pessimo"
                        },
                        {
                            "type": "ColumnSet",
                            "columns": [
                                {
                                    "type": "Column",
                                    "width": "stretch",
                                    "items": [
                                        {
                                            "type": "Image",
                                            "id": "img_pessimo_cinza",
                                            "horizontalAlignment": "Center",
                                            "width": "55px",
                                            "height": "49px",
                                            "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-pessimo-pb.png",
                                            "isVisible": false
                                        }
                                    ],
                                    "selectAction": {
                                        "type": "Action.ToggleVisibility",
                                        "targetElements": [
                                            {
                                                "elementId": "txt_motivo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "motivo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "act_btn_otimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_bom",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_regular",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_ruim",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "act_btn_pessimo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_otimo",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_bom",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_regular",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_ruim",
                                                "isVisible": false
                                            },
                                            {
                                                "elementId": "img_pessimo",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_otimo_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_bom_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_regular_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_ruim_cinza",
                                                "isVisible": true
                                            },
                                            {
                                                "elementId": "img_pessimo_cinza",
                                                "isVisible": false
                                            }
                                        ]
                                    }
                                }
                            ],
                            "spacing": "None"
                        },
                        {
                            "type": "TextBlock",
                            "text": "Péssimo",
                            "wrap": true,
                            "horizontalAlignment": "Center",
                            "color": "Light",
                            "weight": "Bolder"
                        }
                    ],
                    "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "targetElements": [
                            {
                                "elementId": "txt_motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "motivo",
                                "isVisible": true
                            },
                            {
                                "elementId": "act_btn_otimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_bom",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_regular",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_ruim",
                                "isVisible": false
                            },
                            {
                                "elementId": "act_btn_pessimo",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_otimo",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_bom",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_regular",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_ruim",
                                "isVisible": false
                            },
                            {
                                "elementId": "img_pessimo",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_otimo_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_bom_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_regular_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_ruim_cinza",
                                "isVisible": true
                            },
                            {
                                "elementId": "img_pessimo_cinza",
                                "isVisible": false
                            }
                        ]
                    }
                }
            ]
        },
        {
            "type": "TextBlock",
            "text": "O seu comentário ajudará a aprimorar nossos serviços. Informe, por favor, o motivo da sua avaliação.",
            "wrap": true,
            "separator": true,
            "horizontalAlignment": "Center",
            "color": "Light",
            "size": "Medium",
            "weight": "Default",
            "spacing": "Medium",
            "id": "txt_motivo",
            "isVisible": false,
            "required": true
        },
        {
            "type": "Input.Text",
            "placeholder": "Digite uma mensagem",
            "id": "motivo",
            "isVisible": false
        },
        {
            "type": "ColumnSet",
            "columns": [
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": " ",
                            "wrap": true
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.Execute",
                                    "title": "Enviar",
                                    "id": "btn_otimo",
                                    "data": {
                                        "Nota": "Ótimo"
                                    },
                                    "verb": "pesquisaSatisfacao"
                                }
                            ],
                            "id": "act_btn_otimo",
                            "isVisible": false
                        },
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.Execute",
                                    "title": "Enviar",
                                    "id": "btn_bom",
                                    "verb": "pesquisaSatisfacao",
                                    "data": {
                                        "Nota": "Bom"
                                    }
                                }
                            ],
                            "id": "act_btn_bom",
                            "isVisible": false
                        },
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.Execute",
                                    "title": "Enviar",
                                    "id": "btn_regular",
                                    "verb": "pesquisaSatisfacao",
                                    "data": {
                                        "Nota": "Regular"
                                    }
                                }
                            ],
                            "isVisible": false,
                            "id": "act_btn_regular"
                        },
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.Execute",
                                    "title": "Enviar",
                                    "id": "btn_ruim",
                                    "verb": "pesquisaSatisfacao",
                                    "data": {
                                        "Nota": "Ruim"
                                    }
                                }
                            ],
                            "isVisible": false,
                            "id": "act_btn_ruim"
                        },
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.Execute",
                                    "title": "Enviar",
                                    "id": "btn_pessimo",
                                    "verb": "pesquisaSatisfacao",
                                    "data": {
                                        "Nota": "Péssimo"
                                    }
                                }
                            ],
                            "isVisible": false,
                            "id": "act_btn_pessimo"
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "stretch",
                    "items": [
                        {
                            "type": "TextBlock",
                            "text": " ",
                            "wrap": true
                        }
                    ]
                }
            ]
        },
        {
            "type": "Input.Text",
            "id": "input_chamado",
            "value": "${chamado}",
            "isVisible": false
        },
        {
            "type": "Input.Text",
            "id": "input_login",
            "value": "${login}",
            "isVisible": false
        },
        {
            "type": "Input.Text",
            "id": "input_conexao",
            "value": "${conexao}",
            "isVisible": false
        }
    ],
    "backgroundImage": {
        "url": "https://cstibotprod.blob.core.windows.net/imagenschatbottics/03-bg-pesquisa-1.png"
    }
}`

**Comment by 01** [comment]
This is affecting ITSM enterprise chat bot and need guidance on timeline for this to be resolved

**Comment by Cameramorphic** [comment]
@Prasad-MSFT We also encountered this issue yesterday/today. It started to happen at different times on multiple devices, I thought I was going crazy!
This jeopardizes the release of our bot which **must** to be rolled out to our customers at the start of next week!
The bot heavily relies on "ToggleVisibility" and cannot be used properly without it!

**These are our findings / issues we encountered:**
For context, we are using `"selectAction": { "type": "Action.ToggleVisibility" }` on a Container:
<details>
<summary>show simplified sample JSON template</summary>

```json
{
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "type": "AdaptiveCard",
  "version": "1.4",
  "body": [
    {
      "type": "TextBlock",
      "text": "Hello World!",
      "weight": "bolder"
    },
    {
      "type": "Container",
      "selectAction": {
        "type": "Action.ToggleVisibility",
        "targetElements": [
          "details",
          "showDetails",
          "hideDetails"
        ]
      },
      "items": [
        {
          "type": "TextBlock",
          "id": "showDetails",
          "text": "_-> Show details_",
          "color": "accent"
        },
        {
          "type": "TextBlock",
          "id": "hideDetails",
          "text": "_-> Hide details_",
          "color": "accent",
          "isVisible": false
        }
      ]
    },
    {
      "type": "Container",
      "id": "details",
      "isVisible": false,
      "items": [
        {
          "type": "TextBlock",
          "text": "This text was hidden, but now you can see it!"
        }
      ]
    }
  ]
}
```
</details>

It doesn't work in _personal chats_ and _group chats_ on Linux, Windows and Web client. It also stopped working on macOS _just now_.
But in channels in a team it works on all clients.

When inspecting the toggle element in the browser, it shows the following event listeners:
**personal chat** (where it doesn't work):
![Screenshot from 2023-03-03 00-46-06](https://user-images.githubusercontent.com/49117756/222591750-78fa3a63-28d0-4a0b-b62d-6c04cec5773a.png)
The click handler links to this line of code:
![Screenshot from 2023-03-02 23-22-17](https://user-images.githubusercontent.com/49117756/222592432-599a7522-d660-43ed-b42a-dd355d6d4796.png)


**channel** (where it works):
![Screenshot from 2023-03-03 00-47-25](https://user-images.githubusercontent.com/49117756/222591851-59124ecd-39c3-4d86-a006-6a9c9ba6697b.png)
The click handler links to this line of code:
![Screenshot from 2023-03-02 23-23-10](https://user-images.githubusercontent.com/49117756/222592274-c92eaebf-61da-4eb6-ba26-23e9ccc46c44.png)

I hope this helps, please fix the issue ASAP!

**Teams Versions:**
Windows: 1.6.00.1381 (ARM64) (updated 2023-03-02) and 1.6.00.1381 (64-Bit) (updated 2023-02-10)
Linux: 1.5.00.23861 (64-bit)
macOS: 1.5.00.31156 (updated 2023-03-03)

**Comment by Prasad-MSFT** [comment]
Hi Everyone, Apologies. We have checked the status of the issue with engineering team and the issue is fixed and PR has been raised. Currently there is no specific ETA to share, however we will update the thread once we get confirmation from engineering team when its released publicly.

Thanks for understanding.

**Comment by jonbagg** [comment]
HI @Prasad-MSFT, can you please request expedited processing of this change and production rollout?  As seen in this thread, this issue is blocking many production applications.   We've observed prior bug-fix rollouts taking several weeks, which will be very painful for us, and, I'm sure, for others.

**Comment by joshatrezolveai** [comment]
Agreed with @jonbagg - we need to do better here. Not having an ETA is an unacceptable response to my clients. Who can we escalate too outside of this thread?

**Comment by Meghana-MSFT** [comment]
Hi everyone, the PR fix has been merged. The fix will be available for everyone publicly by end of March. Thank you for your understanding.

**Comment by adityarana420** [comment]
> Hi everyone, the PR fix has been merged. The fix will be available for everyone publicly by end of March. Thank you for your understanding.

This is a really critical issue. If the fix is ready, why can't it be pushed into production earlier?

**Comment by zeroshell09** [comment]
@Meghana-MSFT Any workaround ? our customers can no longer  access products list . Really critical issue guyz !

**Comment by andrekiba** [comment]
@Meghana-MSFT same thing for us and our customers, end of March seems something absolutely absurd and not feasible

**Comment by michaelslz23** [comment]
@Meghana-MSFT Same here for many our users and we also have some new features release pending on this bug....Really critical issue!

**Comment by Wajeed-msft** [comment]
@andrekiba, @michaelslz23, @zeroshell09, @adityarana420 - to expedite the fix rollout, could you please provide following information?  

How many apps are impacted by this issue?
How many users are impacted?

**Comment by michaelslz23** [comment]
Several thousand users for me.

**Comment by adityarana420** [comment]
@Wajeed-msft Almost all the apps that use toggleVisibility is affected by this issue. So, that means, all our current apps, all our future apps roll out for march and all the users are impacted.

**Comment by nishanthsekar** [comment]
Hi,

### **I am using an adaptive card with "Action.togglevisibility" property. The adaptive card version is 1.3. The toggle action is working as expected in the teams mobile app(Android and iOS), but it is not working in windows and the web teams application. Need to figure out the issues as it is important.**

Microsoft Teams Version 1.6.00.4472 (64-bit)
AdaptiveCards Nuget Version 3.0.0
Bot Framework SDK v4

**Comment by andrekiba** [comment]
mainly two apps with 300 users each...
but apart from the single case tell me, really you don't have an idea about the impact of this issue???
Toggle visibility is used in pretty much every card...for one reason or another...gosh how many apps are there out there that use cards in Teams??
Really do you think that more than one month to release a fix it si something suitable?
If the answer is yes how can you think that we can propose to a customer to develop an app that is based on this (and other core features of cards in Teams)??

**Comment by 01** [comment]
> mainly two apps with 300 users each... but apart from the single case tell me, really you don't have an idea about the impact of this issue??? Toggle visibility is used in pretty much every card...for one reason or another...gosh how many apps are there out there that use cards in Teams?? Really do you think that more than one month to release a fix it is something suitable? If the answer is yes how can you think that we can propose to a customer to develop an app that is based on this (and other core features of cards in Teams)??

@Meghana-MSFT 
Also agree. I have an enterprise ITSM bot servicing 15k+ employees that large subsection of functionalities have become unusable. We have managed a work around by switching from Toggle Visibility to Show Cards or updating card programmatically to hide show sections  for more critical functionalities, but this fix essentially destroys the UX/UI for those in the mean time. I have already had to sacrifice quite a bit of my teams development bandwidth for these workaround. If it will help expediate this please feel free to have someone contact me directly.

**Comment by nishanthsekar** [comment]
> Hi,
> 
> ### **I am using an adaptive card with "Action.togglevisibility" property. The adaptive card version is 1.3. The toggle action is working as expected in the teams mobile app(Android and iOS), but it is not working in windows and the web teams application. Need to figure out the issues as it is important.**
> Microsoft Teams Version 1.6.00.4472 (64-bit) AdaptiveCards Nuget Version 3.0.0 Bot Framework SDK v4

@Meghana-MSFT @Wajeed-msft

**Comment by Wajeed-msft** [comment]
@nishanthsekar - Fix for this is [being rolled out](https://github.com/MicrosoftDocs/msteams-docs/issues/8157#issuecomment-1459616267). 

@01 , @andrekiba, @adityarana420, @michaelslz23 - Thank you for sharing this information. I'll get back to you once we get updates from engineering team.

**Comment by joshatrezolveai** [comment]
200k minimum impacted here.

**Comment by Wajeed-msft** [comment]
@andrekiba, @michaelslz23, @01,  @zeroshell09, @adityarana420, @joshatrezolveai, @nishanthsekar - could you please share the Ids of tenants which are most impacted so that we could roll out an early fix to these tenants?

**Comment by joseglauber** [comment]
> @andrekiba, @michaelslz23, @01, @zeroshell09, @adityarana420, @joshatrezolveai, @nishanthsekar - could you please share the Ids of tenants which are most impacted so that we could roll out an early fix to these tenants?

I'm with the same problem.
Tenant ID: 2a7030a7-3850-453e-96b7-cd5f58ba9ec2

**Comment by andrekiba** [comment]
Tenant ID: 3ce34e42-c07d-47bb-b72a-4ce606de6b88
thanks

**Comment by nishanthsekar** [comment]
> @andrekiba, @michaelslz23, @01, @zeroshell09, @adityarana420, @joshatrezolveai, @nishanthsekar - could you please share the Ids of tenants which are most impacted so that we could roll out an early fix to these tenants?

@Wajeed-msft could you please share your official mail ID. I will share it over the mail.

**Comment by adityarana420** [comment]
> @andrekiba, @michaelslz23, @01, @zeroshell09, @adityarana420, @joshatrezolveai, @nishanthsekar - could you please share the Ids of tenants which are most impacted so that we could roll out an early fix to these tenants?

Tenant ID: bea78b3c-4cdb-4130-854a-1d193232e5f4

**Comment by Wajeed-msft** [comment]
@nishanthsekar - Please share it on [Microsoft Teams Community Help](mailto:microsoftteamsdev@microsoft.com) alias.

**Comment by michaelslz23** [comment]
@Wajeed-msft  what's the status of this issue and the ETA of the fix?  Thanks.   We're using MS internal tenant

**Comment by Wajeed-msft** [comment]
@joseglauber, @andrekiba, @nishanthsekar, @adityarana420 - This fix is rolled out as part of Teams client version 1.6.00.6754. This build is pinned for tenant Ids shared by you. Please update your Teams client.

**Comment by Wajeed-msft** [comment]
> what's the status of this issue and the ETA of the fix? Thanks. We're using MS internal tenant

@michaelslz23 - This fix should already be available for Microsoft tenant. Please update your Teams client.

**Comment by michaelslz23** [comment]
@Wajeed-msft Thanks.  Confirmed that it's solved in MS internal tenant.

**Comment by Wajeed-msft** [comment]
@b95678 - Fix is rolled out as part of Teams client version 1.6.00.6754. This should be available globally. Could you please update the Teams client and let me know if you are still experiencing this issue. Thanks
